mdtraj/geometry/src/geometry.cpp

```diff
@@ -132,6 +132,82 @@ void dist_mic_triclinic(const float* xyz, const int* pairs, const float* box_mat
     }
 }
 
+void dist_mic_triclinic_t(const float* xyz, const int* pairs, const int* times,
+                        const float* box_matrix, float* distance_out, 
+                        float* displacement_out, const int n_frames,
+                        const int n_atoms, const int n_pairs) {
+    bool store_displacement = (displacement_out != NULL);
+    bool store_distance = (distance_out != NULL);
+    for (int i = 0; i < n_frames; i++) {
+        // Load the periodic box vectors and make sure they're in reduced form.
+        // Get box from frame of first index of time pair
+        int box_offset = times[2*i + 0] * 9;
+        box_matrix += box_offset;
```

Comment:
I don't understand the purpose of this addition and the corresponding subtraction at the end of the function.

Reply:
The goal here is get the box vector information at the first index of `time`. 

The purpose of `box_matrix += box_offset` is to get the box information at the first index of time.  Then, the subtraction at the end of the function is to reset the box information to the first frame.  

 For example, if the distance between atom pairs at `t=0` and `t=2`, the goal is to grab the box information at `t=0`. The `box_offset` allows us to grab the box information at this time, and then will reset to the first frame.   If the next time pair is at `t=5` and `t=7`, `box_offset` will find the box information at `t=5`.

Instead of subtracting the `box_offset` at the end of the function, it may be better to get the correct indices of `box_matrix` by finding the difference between the previous and current time frames of interest.

It's also probably worth noting in the docstrings the the time-dependent calculations are best performed in an  ensemble with constant volume.  In this case, the box information will be constant throughout the course of the trajectory.

Reply:
Reading through it a second time, I understand.

Rather than actually modifying box_offset, you maybe could just index box_matrix by `box_offset + 0` through `box_offset + 8` in the lines below. It might be more clear. But I see.